### ISSUE TITLE
Yup I am misusing the issue tracker

### ISSUE THREAD

**Issue by DaniGuardiola** [SweetAndFluffy/PlayVibe-Integration#1]
Just wanted to say that it is awesome how you mix sex and technology :D
Let me know if I can be of any help (though I only know design and web technologies, but I'm pretty good at it)

**Comment by qdot** [comment]
Yay, someone else is already misusing the issue tracker so I won't feel bad about doing the same!

First off, awesome project! Really cool to see more video game/vibrator projects. I've got a bunch of drivers posted over at my github, @qdot, and the @metafetish org github (I also run http://www.metafetish.com, gonna try to get a post up about this project ASAP), and you might also be interested in the @teledildonics org.

There's an IRC channel for discussion DIY SW/HW too, irc.hackint.org, #teledildonics. Quite a few people in there from all over the world discussing different projects, always looking for new developers!

**Comment by SweetAndFluffy** [comment]
Thank you both for your interest and comments! Don't worry about misusing the issue tracker, it won't break anything :)

@DaniGuardiola I appreciate your offer! I didn't have plans for a web interface yet, but a web based settings page would be nice in my opinion. Something that displays a graph for each vibrator/mode, which can be adjusted by the user to customize the vibration intensity you receive in relation to the amount of points you earned. Do you know what I mean? 

The current interface (a JSON configuration string) is not very user friendly, a web UI would be a real improvement here. I don't know if you'd like to look into the [Flask](http://flask.pocoo.org) framework which is used here, but if you want to make a little front-end for this functionality, I would be very happy to build it into the project! :)

@qdot You have some awesome repositories, I especially like the drivers for toys and the emokit. I'm looking forard to an update on your site! I didn't even think about something like Deldo, this is genius xD

I will look into #teledildonics, sounds quite interesting! We should add some compatibility between our devices, a unified protocol won't hurt anyone I think. Would you be interested in something like that too?

**Comment by qdot** [comment]
Yeah, have been a little slow on updating metafetish lately, need to get back to that. 

BTW, if you're interested in other DIY type boards for sex toy stuff there's:
- Metafetish Pen15 board (I made this years ago, I /think/ it's still on sale?) http://metafetish.com/pen15/
- Orgasmatronics Hackoff (http://www.orgasmatronics.com/products/hackoff) and Master Beta (http://www.orgasmatronics.com/products/master-beta-kit)
- Comingle (http://comingle.io), unfortunately held up by patent lawsuit currently

In terms of unified protocol work, I started work on a framework for this a couple of years ago, that I really need to pick back up:

https://github.com/qdot/buttplug

This is a mess of python using gevent and zeromq. I was able to run 500 (virtual) real touch devices simultaniously off it without even optimizing. There's a half-written design doc for it at 

https://github.com/qdot/buttplug/blob/master/doc/fe-design.org

Frixion (http://frixion.me) is working on the same idea, though they're looking to commercialize and build an app store around it. However, they're wrapped up in the patent lawsuit too, so no clue when that'll be happening.

**Comment by DaniGuardiola** [comment]
@SweetAndFluffy yeah could do that in one day (and I don't have time for more than that), so let's keep talking
Do you wanna provide the web interface from the raspberry? You could just serve it to some port on the local network (to configure it on another device) and make a little rest API I can use.
If the idea is that basic I can do a nice and simple material designed settings page very quickly!

**Comment by DaniGuardiola** [comment]
I don't know about the Flask framework but I'm not very backend-friendly, only used PHP and NodeJS for that before and not so much, so let me know what technology / architecture you prefer to make the interface to communicate with the front-end.

**Comment by SweetAndFluffy** [comment]
Okay, I will implement a rest API to make a configuration change possible. I would prefer the frontend to send commands over normal http requests to the python backend. The interface will look something like `GET /config/` to retrieve the current configuration. `POST /config/` with the JSON of the config as a parameter sets a new config string and reloads the system. 

The configuration string looks like this (just an example). The modes basically define the behavior of the vibrator.  

``` javascript
[
    {
        "modes": [
            {
                "max": 100,
                "min": 0,
                    "multiplicator": 0.5,
                "mode": "ContinuousVibrations"
            },
                {
                    "max": 200,
                    "min": 100,
                    "intensity": 82,
                    "mode": "FixedIntensity"
                },
                {
                    "max": 300,
                    "min": 200,
                    "intensity": 90,
                    "mode": "FixedIntensity"
                },
        ],
        "name": "Default Vibrator",
        "pin": 18,
        "pwm": true
    }
]
```

There is no documentation of the available modes yet, sadly. The best 2 things to implement right now in a settings page would be the "ContinousVibrations" mode and "FixedIntensity". ContinousVibrations means, that the vibrations get more intense, the more points you receive. Fixed means, that as long as you are between the specified maximum and minimum values, you will stay exactly at that setting. I will write some documentation for the system when I have time, so that the mode-system is more clear to people. 

Yes, the idea is basically that basic. I will serve your files from a static web server on the raspberry pi and you can just use the same hostname as your API. I'm looking forward to hear from you again!

**Comment by DaniGuardiola** [comment]
Hey, looks great, can we schedule a day to work on it? I won't need your help most of the time but would be nice to have you online during development. It should be fast.
Please email me: contacto@daniguardiola.me

**Comment by DaniGuardiola** [comment]
:')
